rundown/rundown.py

```diff
@@ -0,0 +1,433 @@
+
+
+class _Auth:
+    """Class supporting creation of RapidAPIAuth and RundownAuth objects."""
+
+    @classmethod
+    def factory(cls, provider: str, api_key: str) -> Union[_RapidAPIAuth, _RundownAuth]:
+        """Factory method returning RapidAPIAuth or RundownAuth object."""
+        if provider not in ["rapidapi", "rundown"]:
+            raise ValueError("API Provider must be either 'rapidapi' or 'rundown'")
+        return (
+            _RapidAPIAuth(api_key) if provider == "rapidapi" else _RundownAuth(api_key)
+        )
+
+
+class Rundown:
+    """The Rundown REST API client class supporting user configuration.
+
+    Args:
+        api_key: The API key to use.
+        api_provider: The API provider. Must be either 'rundown' or 'rapidapi'
+            (case insensitive).
+        timezone: Your preferred timezone.
+            The following formats are accepted:
+            - A str describing a timezone, similar to ‘US/Pacific’, or ‘Europe/Berlin’.
+            - A str in ISO 8601 style, as in ‘+07:00’.
+            - A str, one of the following: ‘local’, ‘utc’, ‘UTC’.
+
+    timezone will be used to format responses from the API.
+
+    Attributes:
+        timezone (str): Your preferred timezone.
+    """
+
+    def __init__(
+        self,
+        api_key: str,
+        api_provider: str = "rapidapi",
+        timezone: str = "local",
+    ):
+        self._auth = _Auth.factory(api_provider.lower(), api_key)
+        self._session = requests.session()
+        self._session.headers.update(self._auth.headers)
+
+        self.timezone = timezone
+
+    def _build_url(self, *segments: Union[str, int]) -> str:
+        """Build URL without query parameters."""
+        base_url = self._auth.api_url
+        return f"{base_url}/{'/'.join([str(s) for s in segments])}"
+
+    def _clean_params(self, **params: Union[str, int, List[Optional[str]], None]):
+        """Disallow parameters that have `None` or empty list as their value."""
+        return {k: v for k, v in params.items() if v}
+
+    def _get(self, url: str, **params: Union[str, int, List[str]]) -> requests.Response:
+        """Make get request."""
+        res = self._session.get(url, params=params)
+        # TODO: handle 404 not found - should never happen if called through methods.
+        return res
+
+    def _build_url_and_get_json(
+        self, *segments: Union[str, int], **params: Union[str, int, List[str]]
+    ) -> Dict:
+        """Build URL from segments and make get request to API."""
+        url = self._build_url(*segments)
+        params = self._clean_params(**params)
+        res = self._get(url, **params)
+        return res.json()
+
+    def _validate_offset(self, offset: int) -> int:
+        """Determine offset by parameter or self.timezone, with parameter precedence."""
+        if offset is None:
+            offset = utc_shift(self.timezone)
+        return offset
+
+    def _get_events(
+        self,
+        sport_id: int,
```

Comment:
it would be great to be able to support a `sport_name` param here as well (ie: `nfl` or `nhl`)